internal/eventstream/eventstream.go

```diff
@@ -0,0 +1,70 @@
+package eventstream
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/alecthomas/types/pubsub"
+)
+
+// EventStream is a stream of events that can be published and subscribed to, that update a materialized view
+type EventStream[T View] interface {
+	Publish(event Event[T]) error
+
+	View() T
+
+	Subscribe() <-chan Event[T]
+}
+
+// StreamView is a view of an event stream that can be subscribed to, without modifying the stream.
+type StreamView[T View] interface {
+	View() T
+
+	Subscribe() <-chan Event[T]
+}
+
+// View is a read-only view of the materialised current state of the event stream.
+type View interface {
```

Comment:
This is equivalent to `any`, so it's redundant. It'd probably be clearer to rename `T View`  to `View any`.